Example/NYTPhotoViewer-Swift/ViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  ViewController.swift
+//  NYTPhotoViewer-Swift
+//
+//  Created by Mark Keefe on 3/20/15.
+//  Copyright (c) 2015 The New York Times. All rights reserved.
+//
+
+import UIKit
+
+class ViewController: UIViewController, NYTPhotosViewControllerDelegate {
+
+    @IBOutlet weak var imageButton : UIButton?
+    private let photos = PhotosProvider().photos
+    private var photosViewController: NYTPhotosViewController
+    
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: NSBundle?) {
+        photosViewController = NYTPhotosViewController(photos: photos as [AnyObject]!)
+        
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+
+        photosViewController.delegate = self
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        photosViewController = NYTPhotosViewController(photos: photos as [AnyObject]!)
+        
+        super.init(coder: aDecoder)
+
+        photosViewController.delegate = self
+    }
+        
+    @IBAction func buttonTapped(sender: UIButton) {
+        photosViewController.delegate = self
+        presentViewController(photosViewController, animated: true, completion: nil)
+        
+        updateImagesOnPhotosViewController(photosViewController, afterDelayWithPhotos: photos)
+    }
+    
+    func updateImagesOnPhotosViewController(photosViewController: NYTPhotosViewController, afterDelayWithPhotos: [ExamplePhoto]) {
+        
+        let delayTime = dispatch_time(DISPATCH_TIME_NOW, 5 * Int64(NSEC_PER_SEC))
+        
+        dispatch_after(delayTime, dispatch_get_main_queue()) {
+            for photo in self.photos {
+                if photo.image == nil {
+                    photo.image = UIImage(named: PrimaryImageName)
+                    photosViewController.updateImageForPhoto(photo)
+                }
+            }
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        let buttonImage = UIImage(named: PrimaryImageName)
+        imageButton?.setBackgroundImage(buttonImage, forState: UIControlState.Normal)
+    }
+    
+    // MARK: - NYTPhotosViewControllerDelegate
+    
+    func photosViewController(photosViewController: NYTPhotosViewController!, handleActionButtonTappedForPhoto photo: NYTPhoto!) -> Bool {
+
+        if UIDevice.currentDevice().userInterfaceIdiom == .Pad {
+            
+            let shareActivityViewController = UIActivityViewController(activityItems: [photo.image], applicationActivities: nil)
+            
+            shareActivityViewController.completionWithItemsHandler = {(activityType: String!, completed: Bool, items: [AnyObject]!, NSError) in
+                if completed {
+                    self.photosViewController.delegate?.photosViewController!(self.photosViewController, actionCompletedWithActivityType: activityType)
+                }
+            }
+
+            shareActivityViewController.popoverPresentationController?.barButtonItem = photosViewController.rightBarButtonItem
```

Comment:
No...thought it would be okay to set the deployment target for the Swift example to be 8+, but if that's too inconsistent with the framework, I can add an alt fix for iOS 7. lmk?